view/src/extras/asynchronous_input.js

```diff
@@ -0,0 +1,101 @@
+import React from "react";
+import TextField from "@material-ui/core/TextField";
+import Autocomplete from "@material-ui/lab/Autocomplete";
+import CircularProgress from "@material-ui/core/CircularProgress";
+import axios from "axios";
+import { authMiddleWare } from "../util/auth";
+
+/**
+ * Creates an asynchronous input that will query database for options
+ * when it is opened by the user (to avoid loading on page reload). 
+ * @param props Propositions passed from parent that determine where the 
+ *              data should come from, how to update the input state in the 
+ *              parent, and other settings of the input 
+ */
+export default function AsynchronousInput(props) {
```

Comment:
Just to aid my own understanding...  when do you write components as classes vs. as functions like this?

Reply:
The current break down is component pages/db objects (or their equivalent form pages in steppers) are traditional components because I wanted to maintain uniformity and it is easier to change groups of states. All of the other components that are not the component/db object pages or related forms are functional components (because hooks and other features are awesome)